mne/preprocessing/interpolate.py

```diff
@@ -153,3 +186,41 @@ def interpolate_bridged_electrodes(inst, bridged_idx):
 
     inst.info['bads'] = bads_orig
     return inst
+
+
+def _find_centroid(ch_pos, group_names):
+    """Compute the centroid position between N electrodes.
+
+    The centroid should be determined in spherical "head" coordinates which is
+    more accurante than cutting through the scalp by averaging in cartesian
+    coordinates.
+
+    A simple way is to average the location in cartesian coordinate, convert
+    to spehrical coordinate and replace the radius with the average radius of
+    the N points in spherical coordinates.
+
+    Parameters
+    ----------
+    ch_pos : OrderedDict
+        The position of all channels in cartesian coordinates.
+    group_names :
+        The name of the N electrodes used to determine the centroid.
+
+    Returns
+    -------
+    pos_centroid : array of shape (3,)
+        The position of the centroid in cartesian coordinates.
+    """
+    cartesian_positions = np.zeros((len(group_names), 3))
+    sphere_positions = np.zeros((len(group_names), 3))
+    for i, ch_name in enumerate(group_names):
+        cartesian_positions[i, :] = ch_pos[ch_name]
+        sphere_positions[i, :] = _cart_to_sph(ch_pos[ch_name])
+    cartesian_pos_centroid = np.average(cartesian_positions, axis=0)
+    sphere_pos_centroid = _cart_to_sph(cartesian_pos_centroid)
+    # average the radius and overwrite it
+    avg_radius = np.average(sphere_positions, axis=0)[0]
+    sphere_pos_centroid[0, 0] = avg_radius
+    # convert back to cartesian
+    pos_centroid = _sph_to_cart(sphere_pos_centroid)[0, :]
+    return pos_centroid
```

Comment:
I defined this in a separate function because I wanted to add a test specifically for this. Maybe this function and the associated test could be moved to a different module? Not sure where would be a good fit..

Reply:
I think this is an okay place for it, we can move if it's useful elsewhere

Reply:
I don't mind, but we will definitely forget it 😄